### ISSUE TITLE
Audio track navigation

### ISSUE THREAD

**Issue by fcoveram** [WordPress/openverse-frontend#1549]
## Problem
The current audio component needs to allow the composite navigation to meet a11y requirements.

### Issues related
- #1297

### Desktop

https://user-images.githubusercontent.com/895819/176517211-ffbad05a-5a8f-489e-bb87-c8126188f434.mp4

|Snackbar. safe area|Overlapping snackbar|
|:---|:---|
|![image](https://user-images.githubusercontent.com/895819/176517276-7e3bf21d-cabb-40f6-8fac-7b2c8736e5c8.png)|![image](https://user-images.githubusercontent.com/895819/176517424-670f2ba7-0766-4574-bb1c-0d612f3a7c43.png)|

### Mobile

<img src="https://user-images.githubusercontent.com/895819/176517546-a86e4a41-e8e2-47fd-8da8-df32ab16bcee.png" width="300">

## Description
Show the focus style of the composite area and display a snackbar element indicating how to interact with the audio track. The snackbar can disappear after a few seconds or have a closing action that reminds the action and no longer appear during the visitor’s session.

Regarding the interaction, I wonder if the screen can scroll down once the navigation reaches the last visible element. Otherwise, the snackbar element will overlap the audio track; like in the example shared above.

I would like your thoughts on the idea.

## Design file
Figma: [Audio track navigation](https://www.figma.com/file/GeG5nSTdPI887CH39jzjzo/Audio-track-navigation?node-id=0%3A1)

**Comment by sarayourfriend** [comment]
> As for closing the notification, [...] it could close after the user's first interaction, like the next time they press a key or click on something.

I think this is the best way. For signed keyboard users who do not use a screen reader, we cannot assume that the length of time it takes for it to be read by the computer is sufficient for someone reading.

**Comment by panchovm** [comment]
I like @zackkrida's idea. Yet I would like to see how others have solved the problem. Meanwhile, the `sm` breakpoint flow should look like the prototype attached below.

Since visitors can control the wave area with the arrow keys on bigger breakpoints, we should do the same on `sm` and the wave area in the global player react to the time change. And to close the player, I like the idea of doing it by pressing `escape` or a similar action that does not depend on focusing on an element.

https://user-images.githubusercontent.com/895819/176781574-83f96355-d0d6-4c0c-a31b-37d29973554b.mp4

> I wonder if the screen can scroll down once the navigation reaches the last visible element. Otherwise, the snackbar element will overlap the audio track; like in the example shared above.

What do you think we need to do in this case?

**Comment by zackkrida** [comment]
@panchovm @sarayourfriend does it make sense to _only_ show the snackbar when the user navigates to an audio track by keyboard? So basically the user expresses 'I want to use my keyboard to navigate here' and then we alert them of the controls. 

Also regarding the global audio player, if the user presses 'play' on one of the audio tracks we can just dismiss the snackbar. I don't see a case where they'd need to be onscreen at the same time. If the user is able to play audio tracks they probably don't need the advice in the snackbar!

**Comment by sarayourfriend** [comment]
Waiting for them to press play or escape seem like good times to dismiss the snack bar :+1: I think also yes, we can use keyboard navigation detection to determine when to show the snackbar. Maybe when an audio row is focused we could see if the link element matches `:focus-visible` selector?

**Comment by panchovm** [comment]
So it seems I can update the final mockups and mark this issue as `ready for work`. I just noticed that label. Should we use it to indicate that design issues reached a done status?

**Comment by zackkrida** [comment]
@panchovm i think removing 'needs design' should be enough! Curious what others think though

**Comment by sarayourfriend** [comment]
We also have the related https://github.com/WordPress/openverse-frontend/issues/1297 issue :thinking:

**Comment by zackkrida** [comment]
Ah that's right, maybe we just close this and add a reference to this and figma to #1297

**Comment by panchovm** [comment]
What if we close #1297 and create the PR from this issue? To me, the design source is clearest from a design issue than a report issue.

However, #1297 has the `I would be interested in implementing this feature` section.

Generally speaking, I like the idea of a workflow going as `report -> design -> implementation`.

**Comment by zackkrida** [comment]
#1297 also has a lot of tech information that would be very relevant to a developer. This particular issues goes beyond adding the new snackbar design, it requires some significant tech changes to the way the audio component works so I'm not sure.

**Comment by panchovm** [comment]
That makes sense. I will close this issue. Thank you all for the iteration ✨

**Comment by zackkrida** [comment]
@panchovm could you add the design references to #1297? Feel free to edit the original post.
